renderer/reducers/settings.js

```diff
@@ -20,12 +23,23 @@ export const UPDATE_CONFIG = 'UPDATE_CONFIG'
  * Fetch the current settings from the database and save into the store.
  */
 export const initSettings = () => async dispatch => {
-  const settings = await window.db.settings.toArray()
-  dispatch({ type: SET_SETTINGS, settings })
+  dispatch({ type: INIT_SETTINGS })
+  try {
+    const allSettings = await window.db.settings.toArray()
+    const settings = allSettings.reduce((obj, item) => {
+      obj[item.key] = item.value
+      return obj
+    }, {})
+    dispatch({ type: INIT_SETTINGS_SUCCESS, settings })
+  } catch (e) {
+    dispatch({ type: INIT_SETTINGS_FAILURE, initSettingsError: e })
+    dispatch(showError(`Unable to load settings: ${e.message}`, { timeout: 0 }))
```

Comment:
This error is not intl compatible. we have a similar problem in a couple of places. Theoretically, we could create a middleware or some other approach to be able to use intl in actions. I'd suggest at least to add `TODO` item here, so these places are easily searchable in future

Reply:
Good call. Have added a `TODO`